pkg/config/store/store2.go

```diff
@@ -165,13 +165,36 @@ func (s *store2) Watch(ctx context.Context) (<-chan Event, error) {
 	return q.chout, nil
 }
 
+const (
+	ruleKind      = "rule"
+	selectorField = "selector"
+	matchField    = "match"
+)
+
+// warnDeprecationAndFix warns users about deprecated fields.
+// It maps the field into new name.
+func warnDeprecationAndFix(key Key, spec map[string]interface{}) map[string]interface{} {
```

Comment:
Why warn...  why not just error out, as bad field ? Else if people ignore warning, they we might end up with configs with selector and match going forward..

Reply:
This code is why integration tests already pass.